### PR TITLE
P1-15: roll out model schemas for AutoGluon families

### PR DESCRIPTION
**Pull request by RobotPsychologist**
## Summary
- add model-config schema + runtime-adapter routes for AutoGluon-backed families:
  - `chronos2`
  - `deepar`
  - `patchtst`
  - `tft`
  - `naive_baseline`
  - `statistical`
- wire `ModelFactory.create_model(...)` paths for those families through `build_model_runtime_config(...)` so runtime payloads are schema-validated
- normalize `learning_rate` alias to `lr` for DeepAR/PatchTST/TFT schema lanes
- preserve PatchTST compatibility with scientific-notation YAML strings like `1e-08`
- expand model schema loader tests (repo config fixtures + factory-path adapter checks)
- regenerate docs-visible JSON schema artifacts + manifest for new model schema classes
- update P1-15 execution plan matrix status for this rollout wave

## Validation
- `ruff check src/config/schemas/model_configs.py src/config/schemas/__init__.py src/config/schemas/json_schema_artifacts.py src/workflows/forecasting/modeling.py tests/workflows/forecasting/test_model_config_schema_loader.py tests/workflows/forecasting/test_config_schema_artifact_generation.py`
- `pytest -q tests/workflows/forecasting/test_model_config_schema_loader.py tests/workflows/forecasting/test_config_schema_artifact_generation.py tests/workflows/forecasting/test_workflow_request_schema.py tests/data/versioning/test_holdout_config_schema_loader.py`
- `python -m src.config.schemas.json_schema_artifacts`
- validated all YAMLs in `configs/models/{chronos2,deepar,naive_baseline,patchtst,statistical,tft}` through `load_model_config_from_yaml(..., model_type=...)` (`validated=81`, `all_passed`)

### CODE REVIEW

**Comment by RobotPsychologist** [comment]
Addressed the review regressions in commit d86cece:

- Removed unsupported `num_epochs` from Chronos2 factory runtime payload (fixes hard failure with `extra=forbid`).
- Restored Chronos2 schema default parity for `covariate_cols` to `['iob']`.
- Updated Chronos2 numeric-string validator to emit field-specific errors (`fine_tune_lr` vs `learning_rate`).
- Added focused Chronos2 regression tests:
  - runtime default parity for covariates,
  - field-specific numeric error messaging,
  - factory-path schema routing (including explicit no-`num_epochs` assertion),
  - factory default-covariate behavior.
- Hardened the model implementation checklist with required default-parity checks and explicit per-family factory-path test coverage.

Validation run on branch:
- `ruff check src/config/schemas/model_configs.py src/workflows/forecasting/modeling.py tests/workflows/forecasting/test_model_config_schema_loader.py`
- `pytest -q tests/workflows/forecasting/test_model_config_schema_loader.py`
- `pytest -q tests/models/test_model_family_contract_suite.py`

File: src/config/schemas/model_configs.py [hunk 1]
```diff
@@ -51,6 +65,196 @@ class TSMixerModelConfigSchema(BaseConfigSchema):
...
+    eval_during_fine_tune: bool = Field(default=True)
+    min_past: int = Field(default=1, gt=0)
+    checkpoint_save_steps: Optional[int] = Field(default=None, gt=0)
+    known_covariate_cols: list[str] = Field(default_factory=list)
+    joint_target_cols: list[str] = Field(default_factory=list)
+
+    @field_validator("learning_rate", "fine_tune_lr", mode="before")
+    @classmethod
+    def _normalize_learning_rate_fields(cls, value: Any) -> Any:
+        return _coerce_numeric_string(value, "learning_rate")
```
[the remark this answers is not in the document]
Comment (RobotPsychologist) [reply src/config/schemas/model_configs.py]: Updated Chronos2 numeric-string validator to emit field-specific errors (fine_tune_lr vs learning_rate)..

File: src/config/schemas/model_configs.py [hunk 2]
```diff
@@ -51,6 +65,196 @@ class TSMixerModelConfigSchema(BaseConfigSchema):
...
+    fine_tune_steps: int = Field(default=15000, gt=0)
+    fine_tune_lr: float = Field(default=1e-5, gt=0.0)
+    fine_tune_batch_size: Optional[int] = Field(default=None, gt=0)
+    batch_size: Optional[int] = Field(default=None, gt=0)
+    fine_tune_logging_steps: Optional[int] = Field(default=None, gt=0)
+    eval_during_fine_tune: bool = Field(default=True)
+    min_past: int = Field(default=1, gt=0)
+    checkpoint_save_steps: Optional[int] = Field(default=None, gt=0)
+    known_covariate_cols: list[str] = Field(default_factory=list)
+    joint_target_cols: list[str] = Field(default_factory=list)
```
[the remark this answers is not in the document]
Comment (RobotPsychologist) [reply src/config/schemas/model_configs.py:132]: Restored Chronos2 schema default parity for covariate_cols to ['iob'].

File: src/workflows/forecasting/modeling.py [hunk 3]
```diff
@@ -154,20 +154,25 @@ def create_model(config: GenericModelConfig):
...
+            extra = dict(config.extra_config) if config.extra_config else {}
+            runtime_config = build_model_runtime_config(
+                model_type=model_type,
+                config_data={
+                    "model_type": "chronos2",
+                    "model_path": config.model_path,
+                    "context_length": config.context_length,
+                    "forecast_length": config.forecast_length,
+                    "batch_size": config.batch_size,
+                    "num_epochs": config.num_epochs,
```
[the remark this answers is not in the document]
Comment (RobotPsychologist) [reply src/workflows/forecasting/modeling.py]: Removed unsupported num_epochs from Chronos2 factory runtime payload (fixes hard failure with extra=forbid).

**Comment by RobotPsychologist** [comment]
Follow-up addressed in commit **53a244d**. I added dedicated factory-path tests for every previously missing schema-routed family so this PR now satisfies checklist §7.2 (`every schema-routed family has at least one dedicated factory-path test`).

### Per-model checklist coverage (models changed in this patch)

#### 1) naive_baseline
- **Factory wiring**: `test_autogluon_family_factory_paths_use_schema_adapter[naive_baseline-default-model-name]`
  - exercises `ModelFactory.create_model(...)` end-to-end via schema adapter
  - validates family default `model_name="Naive"`
- **Unknown field fail-fast**: `test_autogluon_family_factory_paths_report_unknown_runtime_field[naive_baseline]`

#### 2) statistical
- **Factory wiring**: `test_autogluon_family_factory_paths_use_schema_adapter[statistical-default-model-name]`
  - exercises `ModelFactory.create_model(...)` end-to-end via schema adapter
  - validates family default `model_name="AutoARIMA"`
- **Unknown field fail-fast**: `test_autogluon_family_factory_paths_report_unknown_runtime_field[statistical]`

#### 3) patchtst
- **Factory wiring + alias semantics**: `test_autogluon_family_factory_paths_use_schema_adapter[patchtst-learning-rate-alias]`
  - exercises `ModelFactory.create_model(...)` end-to-end via schema adapter
  - verifies `learning_rate` input normalizes to runtime `lr`
- **Unknown field fail-fast**: `test_autogluon_family_factory_paths_report_unknown_runtime_field[patchtst]`

#### 4) tft
- **Factory wiring + alias semantics**: `test_autogluon_family_factory_paths_use_schema_adapter[tft-learning-rate-alias]`
  - exercises `ModelFactory.create_model(...)` end-to-end via schema adapter
  - verifies `learning_rate` input normalizes to runtime `lr`
- **Unknown field fail-fast**: `test_autogluon_family_factory_paths_report_unknown_runtime_field[tft]`

### Validation run
- `ruff check tests/workflows/forecasting/test_model_config_schema_loader.py`
- `pytest -q tests/workflows/forecasting/test_model_config_schema_loader.py` (35 passed)
- `pytest -q tests/models/test_model_family_contract_suite.py` (5 passed)

This closes the checklist gap called out in the review thread.
